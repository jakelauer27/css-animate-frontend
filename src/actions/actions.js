export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const loadPrebuiltAnimations = (prebuiltAnimations) => ({
  type: 'LOAD_PREBUILT_ANIMATIONS',
  prebuiltAnimations
})

export const updateCurrentAnimation = (animation) => ({
  type: 'UPDATE_CURRENT_ANIMATION',
  animation
})

export const saveOriginalAnimation = (animation) => ({
  type: 'SAVE_ORIGINAL_ANIMATION',
  animation
})

export const loadMyAnimations = (myAnimations) => ({
  type: 'LOAD_MY_ANIMATIONS',
  myAnimations
})

export const deleteMyAnimation = (id) => ({
  type: 'DELETE_MY_ANIMATION',
  id
})

export const selectAnimationForEdit = (animation) => ({
  type: 'SELECT_ANIMATION_FOR_EDIT',
  animation
})

export const removeAnimationForEdit = () => ({
  type: 'REMOVE_ANIMATION_FOR_EDIT'
})


