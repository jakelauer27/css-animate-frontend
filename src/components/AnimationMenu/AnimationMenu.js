import React, { Component } from 'react'
import { MyAnimationList } from '../../containers/AnimationList/MyAnimationList';
import { PreBuiltAnimationList } from '../../containers/AnimationList/PreBuiltAnimationList';
import CreateAnimationForm from '../CreateAnimationForm/CreateAnimationForm';


export default class AnimationMenu extends Component {

  render() {
    return (
      <div className='select-ani-container'>
        <div className='infoPopup-overlay'></div>
        <div className='select-ani-popup'>
          <header className='select-ani-container-header'>
            <h2 className='my-ani-header'>My Animations</h2>
            <h2 className='prebuilt-ani-header'>Prebuilt Templates</h2>
          </header>
          <div className='list-container'>
            <MyAnimationList />
            <PreBuiltAnimationList />
          </div>
          <button className='close-select-ani-btn'>Go!</button>
        </div>
        {/* <CreateAnimationForm /> */}
      </div>
    )
  }
}

