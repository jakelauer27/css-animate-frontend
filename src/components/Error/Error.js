import React, { Component } from 'react'
import { Link } from 'react-router-dom'

class Error extends Component {
  
  render() {
    return (
      <div className='error-page'>
        <h1 className='error-main-message'>OOPS!</h1>
        <h2 className='error-secondary-message'>page not found</h2>
        <Link to='/' className='return-to-site-link lower-btn'>return to site</Link>
      </div>
    )
  }
}

export default Error