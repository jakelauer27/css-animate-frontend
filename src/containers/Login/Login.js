import React, { Component } from 'react'
import { Redirect, Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'
import { loginUser } from '../../thunks/login'
import { getMyAnimations } from '../../thunks/getMyAnimations'

export class Login extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
      incorrect: false
    }
  }

  handleChange = (e) => {
    const { value, name } = e.target
    this.setState({
      [name]: value
    })
  }

  async handleSubmit(e) {
    e.preventDefault()
    const { getMyAnimations, loginUser } = this.props
    const { email, password } = this.state
    const success = await loginUser({email, password})
    if(success.status === 'success') {
      await getMyAnimations(success.data.id)
    } else {
      this.setState({incorrect: true})
    }
  }

  render() {
    const { email, password } = this.state
    const { user } = this.props
    if( user.name ) {
      return (
        <Redirect to='/properties' />
      )
    } 
    
    return (
      <div className='login-container'>
        <div className='infoPopup-overlay'></div>
        <div className='form-container'>
          <h2 className='login-title'>Login</h2>
          <form className='login-form' onSubmit={(e) => this.handleSubmit(e)}>
            <input 
              className='form-input email'
              type='text' 
              placeholder='email' 
              value={email} 
              name='email'
              onChange={this.handleChange}>
            </input>
            <input 
              className='form-input password'
              type='password' 
              placeholder='password' 
              value={password}
              name='password'
              onChange={this.handleChange}>
            </input>
            <button>Login</button>
          </form>
          <div className='new-account-container'>
            <h3 className={`incorrect-login ${this.state.incorrect && 'incorrect'}`}>username or password is incorrect</h3>
            <h3>Don't have an account?</h3>
            <Link className='form-link' to='/properties/signup' >Create New Account</Link>
            <Link className='form-link continue-as-guest' to='/properties' >Continue as Guest</Link>
          </div>
        </div>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({ 
  loginUser: (user) => dispatch(loginUser(user)),
  getMyAnimations: (id) => dispatch(getMyAnimations(id))
})

export const mapStateToProps = (state) => ({ 
  user: state.user
})

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  user: PropTypes.object,
  getMyAnimations: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(Login)