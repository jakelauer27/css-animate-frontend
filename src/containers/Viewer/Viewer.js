import React, { Component } from 'react'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'

export class Viewer extends Component {
  constructor() {
    super()
    this.state = {
      animation: {},
      play: true
    }
  }
  
  playAnimation() {
    const { currentAnimation } = this.props
    this.setState({
      animation: {
        animation: `${currentAnimation.name} ${currentAnimation.duration} ${currentAnimation.timingFunction} ${currentAnimation.delay} ${currentAnimation.iterationCount} ${currentAnimation.direction}`,
        animationFillMode: `${currentAnimation.fillMode}`
      },
      play: false
    })
  }

  resetAnimation(waitTime) {
    setTimeout( 
      this.setState({
        animation: {},
        play: true
      })
    , waitTime)
  }

  render() {
    const { currentAnimation } = this.props
    if  (!currentAnimation) {
      return <div></div>
    }
    return (
      <div className='viewer-component'>
        <h2 className='current-animation-label'>*{currentAnimation.name}*</h2>
        <div className='viewer-container'>
          <div className='viewer'>
            <div className='square square-0'
                  style={this.state.animation}
                  onAnimationEnd={() => this.resetAnimation(1000)}>
                <h4 className='main-title'>ani<span>Mate</span></h4>
            </div>
          </div>
        </div>
        <div className='editor-bottom-btns-container'>
          <button className={`lower-btn play-btn`} 
            onClick={() => this.playAnimation()}
            disabled={!this.state.play}>play</button>
          <button className={`lower-btn stop-btn`} 
            onClick={() => this.resetAnimation(0)}
            disabled={this.state.play}>stop</button>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  currentAnimation: state.currentAnimation.properties
})

Viewer.propTypes = {
  currentAnimation: PropTypes.object
}

export default connect(mapStateToProps)(Viewer)