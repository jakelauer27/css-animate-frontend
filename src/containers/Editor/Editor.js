import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Route, Switch, NavLink, Link } from 'react-router-dom'
import KeyframesEditor from '../KeyframesEditor/KeyframesEditor'
import PropertiesEditor from '../PropertiesEditor/PropertiesEditor'
import { updateCurrentAnimation } from '../../actions/actions'
import * as CSSInsertion from '../../utils/keyframesInsertion'
import CopyPopup from '../CopyPopup/CopyPopup'
import { PropTypes } from 'prop-types'

export class Editor extends Component {

  resetAnimation(animation) {
    animation = JSON.parse(animation)
    this.props.updateCurrentAnimation(animation)
    CSSInsertion.updateKeyframes(animation.keyframes)
    this.resetInputValidation()
  }
  
  resetInputValidation() {
    document.querySelectorAll('input').forEach( input => {
      input.classList.remove('red');
    });
    document.querySelector('.play-btn').removeAttribute('disabled');
  }


  render() {
    const { currentAnimation } = this.props

    return (
      <div className='editor-component'>
        <div className='editor-container'>
          <div className='editor-options'>
            <NavLink 
              to={`/properties`}
              className='animation-btn'
              activeClassName='selected'>properties
            </NavLink>
            <NavLink 
              to={`/keyframes`}
              className='keyframes-btn'
              activeClassName='selected'>keyframes
            </NavLink>
          </div>
          <div className='editor'>
          {
          <Switch>
            <Route path={`/properties`} component={PropertiesEditor}/>
            <Route path={`/keyframes`} component={KeyframesEditor}/>
          </Switch>
          }
          </div>
        </div>
        <div className='editor-bottom-btns-container'>
          <Link to={`/properties/copy`} className='lower-btn copy-btn'>
            copy code   
          </Link>
          <Route path={`/properties/copy`} render={() => <CopyPopup currentPage={currentAnimation}/>} />
          <button className={`lower-btn reset-btn`} onClick={() => this.resetAnimation(this.props.originalAnimation)}>
            reset
          </button>
        </div>
      </div>
    )
  }
}


export const mapStateToProps = (state) => ({
  originalAnimation: state.originalAnimation,
  currentAnimation: state.currentAnimation
})

export const mapDispatchToProps = (dispatch) => ({
  updateCurrentAnimation: (animation) => dispatch(updateCurrentAnimation(animation))
})

Editor.propTypes = {
 originalAnimation: PropTypes.string,
 currentAnimation: PropTypes.object,
 updateCurrentAnimation: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(Editor)
