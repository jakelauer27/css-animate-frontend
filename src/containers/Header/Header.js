import React, { Component } from 'react'
import { Link, withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'
import { signOut } from '../../actions/actions'

export class Header extends Component {

  render() {
    const { user, signOut } = this.props
    let greeting = ''
    let login = 'Login'
    if (user.name) {
      login = 'Logout'
      greeting = `Hi ${user.name}!`
    }

    return (
      <header>
        <h1 className='main-title'>CSS ani<span>Mate</span></h1>
        <div className='header-btn-container'>
        <Link to={`properties/selectAnimation`}>
          <button className='load-example-btn'>Load Animation</button>
        </Link>
        </div>
        <div className='inst-log-container'>
          <h3 className='user-greeting'>{greeting}</h3>
          <Link to={`properties/howto/general`}>
            <button className='questions-btn'>Instructions</button>
          </Link>
          <Link to='properties/login'>
            <button className='login-btn' onClick={() => signOut()}>{login}</button>
          </Link>
        </div>
      </header>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  signOut: () => dispatch(signOut())
})

Header.propTypes = {
  user: PropTypes.object,
  signOut: PropTypes.func.isRequired
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header))