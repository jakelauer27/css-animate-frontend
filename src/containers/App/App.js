import React, { Component } from 'react'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'
import { Route, Switch, withRouter, Redirect } from 'react-router-dom'
import { getPrebuiltAnimations } from '../../thunks/getPrebuiltAnimations'
import { getMyAnimations } from '../../thunks/getMyAnimations'
import { updateCurrentAnimation, saveOriginalAnimation } from '../../actions/actions'
import { routes } from '../../utils/routes'
import * as CSSInsertion from '../../utils/keyframesInsertion'
import AnimationMenu from '../AnimationMenu/AnimationMenu'
import Editor from '../Editor/Editor'
import Error from '../../components/Error/Error'
import Header from '../Header/Header'
import HowToPopup from '../../components/HowToPopup/HowToPopup'
import Login from '../Login/Login'
import SignUp from '../SignUp/SignUp'
import Viewer from '../Viewer/Viewer'
import '../../styles/app.scss'

export class App extends Component {
  
  async componentDidMount() {
    const { getPrebuiltAnimations, user_id, getMyAnimations } = this.props
    await getPrebuiltAnimations()
    this.loadInitialAnimation()
    if(user_id) {
      await getMyAnimations(user_id)
    }
  }

  loadInitialAnimation() {
    const { updateCurrentAnimation, prebuiltAnimations, saveOriginalAnimation } = this.props
    updateCurrentAnimation({...prebuiltAnimations[0]})
    saveOriginalAnimation(JSON.stringify({...prebuiltAnimations[0]}))
    CSSInsertion.updateKeyframes({...prebuiltAnimations[0].keyframes}) 
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Switch>
          <Route exact path='/' render={() => {
            return <Redirect to='/properties'/>
          }} />
          <Route path={`/properties`}
          render={() => {
            return ( 
              <main>
                <Editor location={this.props.location}/>
                <Viewer /> 
              </main>
            )
            }}/>   
          <Route path={`/keyframes`}
          render={() => {
            return ( 
              <main>
                <Editor location={this.props.location}/>
                <Viewer /> 
              </main>
            )
            }}/>   
          <Route path={'/error'} component={Error} />
        </Switch>
        <Route path={`/properties/howto`} component={HowToPopup}/>
        <Route path={`/properties/login`} component={Login}/>
        <Route path={`/properties/signup`} component={SignUp}/>
        <Route path={`/properties/selectAnimation`} component={AnimationMenu} />
        <Route exact path='*' render={() => {
          if(!routes.find(route => route === this.props.location.pathname)) {
            return <Redirect to='/error'/>  
          } else return null
        }} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  prebuiltAnimations: state.prebuiltAnimations,
  user_id: state.user.id
})

export const mapDispatchToProps = (dispatch) => ({
  getPrebuiltAnimations: () => dispatch(getPrebuiltAnimations()),
  updateCurrentAnimation: (animation) => dispatch(updateCurrentAnimation(animation)),
  saveOriginalAnimation: (animation) => dispatch(saveOriginalAnimation(animation)),
  getMyAnimations: (id) => dispatch(getMyAnimations(id))
})

App.propTypes = {
  prebuiltAnimations: PropTypes.array,
  user_id: PropTypes.number,
  getPrebuiltAnimations: PropTypes.func.isRequired,
  updateCurrentAnimation: PropTypes.func.isRequired,
  saveOriginalAnimation: PropTypes.func.isRequired,
  getMyAnimations: PropTypes.func.isRequired
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App))

