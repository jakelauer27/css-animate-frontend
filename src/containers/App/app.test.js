import { App, mapStateToProps, mapDispatchToProps } from './App'
import React from 'react'
import { shallow, mount } from 'enzyme'
import { getPrebuiltAnimations } from '../../thunks/getPrebuiltAnimations'
import { getMyAnimations } from '../../thunks/getMyAnimations'
import { updateCurrentAnimation, saveOriginalAnimation } from '../../actions/actions'
import * as CSSInsertion from '../../utils/keyframesInsertion'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import rootReducer from '../../reducers/index'
import AnimationMenu from '../AnimationMenu/AnimationMenu'
import Editor from '../Editor/Editor'
import Error from '../../components/Error/Error'
import Header from '../Header/Header'
import HowToPopup from '../../components/HowToPopup/HowToPopup'
import Login from '../Login/Login'
import SignUp from '../SignUp/SignUp'
import Viewer from '../Viewer/Viewer'
import { createStore } from 'redux'

jest.mock('../../components/HowToPopup/HowToPopup')
jest.mock('../Viewer/Viewer')
jest.mock('../../components/Error/Error')
jest.mock('../Header/Header')
jest.mock('../Editor/Editor')
jest.mock('../Login/Login')
jest.mock('../SignUp/SignUp')
jest.mock('../AnimationMenu/AnimationMenu')
jest.mock('../../utils/keyframesInsertion')
jest.mock('../../thunks/getMyAnimations')
jest.mock('../../thunks/getPrebuiltAnimations')

describe('App', () => {
  let wrapper
  let mockGetPrebuiltAnimations
  let mockUpdateCurrentAnimation
  let mockSaveOriginalAnimation
  let mockGetMyAnimations
  let mockPrebuiltAnimations

  beforeEach(() => {
    mockGetPrebuiltAnimations = jest.fn()
    mockUpdateCurrentAnimation = jest.fn()
    mockSaveOriginalAnimation = jest.fn()
    mockGetMyAnimations = jest.fn()
    mockPrebuiltAnimations = [{name: 'mockAnimation', keyframes: {}}]
    wrapper = shallow( 
      <App 
        prebuiltAnimations={mockPrebuiltAnimations}
        user_id={1}
        getPrebuiltAnimations={mockGetPrebuiltAnimations}
        updateCurrentAnimation={mockUpdateCurrentAnimation}
        saveOriginalAnimation={mockSaveOriginalAnimation}
        getMyAnimations={mockGetMyAnimations}
      />
    )
  })


  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentDidMount', () => {
    
    it('should call getPrebuiltAnimations', async () => {
      await wrapper.instance().componentDidMount()

      expect(mockGetMyAnimations).toHaveBeenCalled()
    })

    it('should call thisInitialAnimation', async () => {
      const spy = jest.spyOn(wrapper.instance(), 'loadInitialAnimation')
      await wrapper.instance().componentDidMount()

      expect(spy).toHaveBeenCalled()
    })

    it('should call getMyAnimations if a user is logged in', async () => {
      await wrapper.instance().componentDidMount()

      expect(mockGetMyAnimations).toHaveBeenCalled()
    })
  })

  describe('loadInitialAnimation', () => {

    it('should call updateCurrentAnimation with the correct params', () => {
      wrapper.instance().loadInitialAnimation()
      expect(mockUpdateCurrentAnimation).toHaveBeenCalledWith(mockPrebuiltAnimations[0])
    })

    it('should call saveOriginalAnimation with the correct params', () => {
      wrapper.instance().loadInitialAnimation()
      expect(mockSaveOriginalAnimation).toHaveBeenCalledWith(JSON.stringify(mockPrebuiltAnimations[0]))

    })

    it('should call CSSinsertion.updateKeyframes with the correct params', () => {
      wrapper.instance().loadInitialAnimation()
      expect(CSSInsertion.updateKeyframes).toHaveBeenCalledWith(mockPrebuiltAnimations[0].keyframes)
    })
  })
  
  describe('mapStateToProps', () => {

    it('shoudld return a props object with a user_id and prebuiltanimations property', () => {
      const mockState = {prebuiltAnimations: {keyframes: {name: 'slideInX'}}, user:{id: 1}}
      const expected = {prebuiltAnimations: {keyframes: {name: 'slideInX'}}, user_id: 1}
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {

    it('calls dispatch with a getPrebuiltAnimations action when getPrebuiltAnimations is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = getPrebuiltAnimations()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.getPrebuiltAnimations()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with a updateCurrentAnimation action when updateCurrentAnimation is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = updateCurrentAnimation()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.updateCurrentAnimation()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with a saveOriginalAnimation action when saveOriginalAnimation is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = saveOriginalAnimation()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.saveOriginalAnimation()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with a getMyAnimations action when getMyAnimations is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = getMyAnimations()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.getMyAnimations()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('routes', () => {

    // it('should redirect to properties if at the base route', () => {
    //   let wrapper = mount(
    //     <Provider store={createStore(() => ({name: 'mockAnimation', keyframes: {}}))}>
    //       <MemoryRouter initialEntries={['/']}>
    //         <App 
    //           prebuiltAnimations={mockPrebuiltAnimations}
    //           user_id={1}
    //           getPrebuiltAnimations={mockGetPrebuiltAnimations}
    //           updateCurrentAnimation={mockUpdateCurrentAnimation}
    //           saveOriginalAnimation={mockSaveOriginalAnimation}
    //           getMyAnimations={mockGetMyAnimations}
    //         />
    //       </MemoryRouter>
    //     </Provider>
    //     )
    //     expect(wrapper.find(Editor)).toHaveLength(1)
    //     expect(wrapper.find(Viewer)).toHaveLength(1)
    // })

    // it('should render Editor and Viewer with the correct animation', () => {
    //   let wrapper = mount(
    //     <Provider store={createStore(() => ({animation: mockAnimation}))}>
    //       <MemoryRouter initialEntries={['/slideInY/properties']}>
    //         <App />
    //       </MemoryRouter>
    //     </Provider>
    //     )
    //     expect(wrapper.find(Editor)).toHaveLength(1)
    //     expect(wrapper.find(Viewer)).toHaveLength(1)
    // })

    // it('should render the howToPopup if at the howto route', () => {
    //   let wrapper = mount(
    //     <Provider store={createStore(() => ({animation: mockAnimation}))}>
    //       <MemoryRouter initialEntries={['/slideInX/properties/howto']}>
    //         <App  />
    //       </MemoryRouter>
    //     </Provider>
    //     )
    //     expect(wrapper.find(HowToPopup)).toHaveLength(1)
    // })
  })
})