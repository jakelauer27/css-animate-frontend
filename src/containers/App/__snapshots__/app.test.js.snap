// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    getMyAnimations={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getPrebuiltAnimations={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    prebuiltAnimations={
      Array [
        Object {
          "keyframes": Object {},
          "name": "mockAnimation",
        },
      ]
    }
    saveOriginalAnimation={
      [MockFunction] {
        "calls": Array [
          Array [
            "{\\"name\\":\\"mockAnimation\\",\\"keyframes\\":{}}",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    updateCurrentAnimation={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "keyframes": Object {},
              "name": "mockAnimation",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    user_id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Header)) />,
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            path="/properties"
            render={[Function]}
          />
          <Route
            path="/keyframes"
            render={[Function]}
          />
          <Route
            component={[MockFunction]}
            path="/error"
          />
        </Switch>,
        <Route
          component={[MockFunction]}
          path="/properties/howto"
        />,
        <Route
          component={[MockFunction]}
          path="/properties/login"
        />,
        <Route
          component={[MockFunction]}
          path="/properties/signup"
        />,
        <Route
          component={[MockFunction]}
          path="/properties/selectAnimation"
        />,
        <Route
          exact={true}
          path="*"
          render={[Function]}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [MockFunction],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              path="/properties"
              render={[Function]}
            />,
            <Route
              path="/keyframes"
              render={[Function]}
            />,
            <Route
              component={[MockFunction]}
              path="/error"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/properties",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/keyframes",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [MockFunction],
              "path": "/error",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [MockFunction],
          "path": "/properties/howto",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [MockFunction],
          "path": "/properties/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [MockFunction],
          "path": "/properties/signup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [MockFunction],
          "path": "/properties/selectAnimation",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "*",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Header)) />,
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/properties"
              render={[Function]}
            />
            <Route
              path="/keyframes"
              render={[Function]}
            />
            <Route
              component={[MockFunction]}
              path="/error"
            />
          </Switch>,
          <Route
            component={[MockFunction]}
            path="/properties/howto"
          />,
          <Route
            component={[MockFunction]}
            path="/properties/login"
          />,
          <Route
            component={[MockFunction]}
            path="/properties/signup"
          />,
          <Route
            component={[MockFunction]}
            path="/properties/selectAnimation"
          />,
          <Route
            exact={true}
            path="*"
            render={[Function]}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [MockFunction],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                path="/properties"
                render={[Function]}
              />,
              <Route
                path="/keyframes"
                render={[Function]}
              />,
              <Route
                component={[MockFunction]}
                path="/error"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/properties",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/keyframes",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [MockFunction],
                "path": "/error",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [MockFunction],
            "path": "/properties/howto",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [MockFunction],
            "path": "/properties/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [MockFunction],
            "path": "/properties/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [MockFunction],
            "path": "/properties/selectAnimation",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "*",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
